pulser/simulation/simulation.py

```diff
@@ -179,15 +187,17 @@ def _build_basis_and_op_matrices(self):
             basis = ['r', 'g', 'h']
             projectors = ['gr', 'hg', 'rr', 'gg', 'hh']
 
-        self.basis = {b: qutip.basis(self.dim, i) for i, b in enumerate(basis)}
+        self.basis = {b: qutip.basis(self.dim, i)
+                      for i, b in enumerate(basis)}
         self.op_matrix = {'I': qutip.qeye(self.dim)}
 
         for proj in projectors:
             self.op_matrix['sigma_' + proj] = (
                 self.basis[proj[0]] * self.basis[proj[1]].dag()
             )
 
-    def _build_operator(self, op_id, *qubit_ids, global_op=False):
+    def _build_operator(self, op_id: str, *qubit_ids: List[str],
```

Comment:
Each qubit ID is either `str` or `int`, so I'd put this as:
```suggestion
    def _build_operator(self, op_id: str, *qubit_ids: Union[str, int],
```